charts/metering-ansible-operator/templates/crds/prestotable.crd.yaml

```diff
@@ -21,3 +21,132 @@ spec:
+            catalog:
+              type: string
+              description: |
+                Catalog specifies which catalog the Presto table is to be created within.
+                In many cases, the catalog will be set to "hive".
+              minLength: 1
+            schema:
+              type: string
+              minLength: 1
+            tableName:
+              type: string
+              description: |
+                TableName is the desired name of the table to be created in Presto,
+                or in the case where "unmanaged" is set to false, the name of an existing table within Presto.
+              minLength: 1
+            columns:
+              type: array
+              description: |
+                Columns is a list of columns that match the schema of the PrestoTable.
+              items:
+                type: object
+                required:
+                - name
+                - type
+                properties:
+                  name:
+                    type: string
+                    minLength: 1
+                  type:
+                    type: string
+                    minLength: 1
+            properties:
+              type: array
+              description: |
+                Properties is a map containing string key and value pairs. Each key-value pair is a table property for
+                configuring the table. The available properties depend on the "catalog" being used.
+                Note: this is an optional field.
+            comment:
+              type: string
+              description: |
+                Comment sets a comment on the Presto table. This can be used to store arbitrary information about a table.
+                Note: this is an optional field.
+              minLength: 1
+            view:
+              type: boolean
+              description: |
+                View controls whether the reporting-operator needs to create a view within Presto.
+                If true, the reporting-operator uses the "query" field as the SELECT statement for creating the view.
+                Note: this is an optional field.
+            createTableAs:
+              type: boolean
+              description: |
+                CreateTableAs controls whether the reporting-operator needs to create a table within Presto.
+                If true, the reporting-operator uses the "query" field as the SELECT statemtnt for creating the table.
+                Note: this is an optional field.
+            query:
+              type: string
+              description: |
+                Query is a string containing a SQL SELECT query used for creating a table or view.
+                Note: this is an optional field.
+              minLength: 1
+          oneOf:
+          - required:
+            - createTableAs
+            - query
+            properties:
+              createTableAs:
+                enum:
+                - true
+              unmanaged:
```

Comment:
I see a few cases of enums with one value. Is enum just for one item?

Reply:
It's just restricting the possible values for the `createTableAs` property in this sub-schema. If I only want `createTableAs: true` to be the only valid value, then I would only specify a single value for `enum`.

See: https://swagger.io/docs/specification/data-models/enums/